src/core/thread/mle_router.cpp

```diff
@@ -264,11 +264,6 @@ ThreadError MleRouter::HandleDetachStart(void)
         mRouters[i].mState = Neighbor::kStateInvalid;
     }
 
```

Comment:
I agree this is something to think about, but I'm not sure on what the proper solution is yet.  For example, if calls to `SetRouterRoleEnabled(false)` then `SetRouterRoleEnable(true)` happen in close succession, should the Router try to resume its child connections?

I'm going to merge this now since it fixes a known bug.  We can look to improve on this later.  Thanks.